__tests__/analyzer/schema-generator.js

```diff
@@ -0,0 +1,20 @@
+const { join } = require('path');
+const { generateSchema } = require('../../src/analyzer/schema-generator');
+const { FileSystemFileBrowserAdapter } = require('../../src/analyzer/file-browser');
+const emptyRepoPath = join(__dirname, '../data/empty-repo');
+const azimuthRepoPath = join(__dirname, '../data/azimuth');
+
+describe('generate schema', () => {
+    test('empty repo', async () => {
+        const fileBrowserAdapter = new FileSystemFileBrowserAdapter({ dirPath: emptyRepoPath });
+        const schemaGeneratorResult = await generateSchema({ ssgMatchResult: null, fileBrowserAdapter });
+        expect(schemaGeneratorResult).toBeNull();
+    });
+
+    // todo make it works with autogenerated ids
+    test.skip('azimuth', async () => {
+        const fileBrowserAdapter = new FileSystemFileBrowserAdapter({ dirPath: azimuthRepoPath });
+        const schemaGeneratorResult = await generateSchema({ ssgMatchResult: null, fileBrowserAdapter });
+        expect(schemaGeneratorResult).toBe(azimuthGeneratedResult);
```

Comment:
it's hard to stub `schemaGeneratorResult` for real data because ids are prefixed with random hash, @smnh do you think we can omit them or just do some other workaround?

Reply:
Don't try to test the whole result at once, this test will break as soon as something is changed/improved.
Test something specific, look how other tests are written. Look how config-loader.js and config-loader.js tests are written. They never compare full results.